CorsixTH/Src/th_gfx.h

```diff
@@ -518,21 +518,32 @@ class animation_base : public drawable {
   void attach_to_tile(map_tile* pMapNode, int layer);
 
   uint32_t get_flags() const { return flags; }
-  int get_x() const { return x_relative_to_tile; }
-  int get_y() const { return y_relative_to_tile; }
+  int get_x_pixels() const { return x_tile_pixels; }
+  int get_y_pixels() const { return y_tile_pixels; }
+  int get_x_tile() const { return x_tile; }
+  int get_y_tile() const { return y_tile; }
 
   void set_flags(uint32_t iFlags) { flags = iFlags; }
-  void set_position(int iX, int iY) {
-    x_relative_to_tile = iX, y_relative_to_tile = iY;
+  void set_pixel_position(int x, int y) {
+    x_tile_pixels = x;
+    y_tile_pixels = y;
+  }
+  void set_tile_position(int x, int y) {
+    x_tile = x;
+    y_tile = y;
   }
   void set_layer(int iLayer, int iId);
   void set_layers_from(const animation_base* pSrc) { layers = pSrc->layers; }
 
  protected:
-  //! X position on tile (not tile x-index)
-  int x_relative_to_tile{};
-  //! Y position on tile (not tile y-index)
-  int y_relative_to_tile{};
+  //! X tile position. Negative value indicates the animation is not active.
+  int x_tile{-1};
+  //! Y tile position. Negative value indicates the animation is not active.
+  int y_tile{-1};
+  //! X pixel position relative to animation tile center.
+  int x_tile_pixels{};
```

Comment:
I find these names less clear than the old ones, but they seem to serve the same purpose (x_tile_pixels seems like it could be the width, or the position of the tile).

Maybe
```
struct pixel_offset {int x{}, int y{}};
...
pixel_offset tile_center_offset;
```

Reply:
Or pixel_diff, like the ptrdiff_t type